### ISSUE TITLE
onSearch function not being called

### ISSUE THREAD

**Issue by cameeob2003** [eakoriakin/ionic-select-searchable#18]
Im having an issue where the onSearch method defined in the controller is not being called (might be something stupid on my part). 

My code is the following:

```
<ion-item>
    <select-searchable
        title="Currencies"
        [items]="currencyOptions"
        itemValueField="id"
        itemTextField="name"
        (onSearch)="searchCurrencies($event)"
        [canSearch]="true"
        [hasInfiniteScroll]="true"
        (onInfiniteScroll)="getMoreCurrencies($event)"
        (onChange)="currencyChange($event)">
    </select-searchable>
  </ion-item>
```

```
Component({
  selector: 'page-watchlist',
  templateUrl: 'watchlist.html'
})
export class WatchlistPage extends BaseComponent implements OnInit {
  ...
  searchCurrencies(event: { component: SelectSearchable, text: string }) {
      event.component.isSearching = true;
      console.log(event.text);
      let text = (event.text || '').trim().toLowerCase();

      if (!text) {
          event.component.items = [];
          return;
      } else if (event.text.length < 2) {
          return;
      }

      this.socket.sendChannelMessage("currency:search", {"text": text}, (resp) => {
        console.log(resp);
        event.component.items = resp.data;
        event.component.isSearching = false;
      });
  }
  ...
}
```

The onInfiniteScroll & onChange functions fire with no issue. I cant get the onSearch function to fire for the life of me. I haven't delved into the lib code to deeply but will in the AM.

**Comment by eakoriakin** [comment]
Hey, I've fixed it in `1.0.13` version. Let me know if it works for you.
Also, you might wanna to change `searchCurrencies`, so when `text` is empty all items are shown.

```
  searchCurrencies(event: { component: SelectSearchable, text: string }) {
      ...
      if (!text) {
          // Assign all items here.
          event.component.items = allCurrencies;
          return;
      } else if (event.text.length < 2) {
          return;
      }
      ...
  }
```

**Comment by cameeob2003** [comment]
Great response time! It's all working now. I appreciate the help on this one, I'll close the issue.
